tests/ztest/error_hook/src/main.c

```diff
@@ -41,15 +41,25 @@ static void trigger_assert_fail(void *a)
 	__ASSERT(a != NULL, "parameter a should not be NULL!");
 }
 
-static void trigger_fault_illegal_instruction(void)
+/*
+ * Do not optimize to prevent GCC from generating invalid
+ * opcode exception instruction instead of real instruction.
+ * See #33403
```

Comment:
Do we want to mention issue number here? @nashif I thought in the past I was advised against this, please correct me if I'm wrong.

Reply:
Excuse me, @jenmwms  I did not know about this rule. I removed this issue number. Thanks for your reminder.